core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1014,6 +1028,11 @@ public void onFailure(String source, Exception e) {
             }
 
             @Override
+            public void onNoLongerMaster(String source) {
+                listener.onNoLongerMaster(source);
```

Comment:
can listener be null here? It is marked as Nullable above?

Reply:
Yes! Good catch! Fixing it.